cli/flox-rust-sdk/src/models/environment/remote_environment.rs

```diff
@@ -268,6 +269,15 @@ impl Environment for RemoteEnvironment {
         Ok(self.out_link.clone())
     }
 
+    /// Return a path that environment hooks should use to store transient data.
+    ///
+    /// Remote environments shouldn't have state of any kind, so this just
+    /// returns a temporary directory.
+    fn cache_path(&self) -> Result<PathBuf, EnvironmentError2> {
+        let tempdir = TempDir::new().map_err(EnvironmentError2::CreateTempDir)?;
+        Ok(tempdir.into_path())
```

Comment:
suggestion: use asociated .flox dir
Remote environments are based on a `ManagedEnvironment` in `FLOX_CACHE_DIR/owner/name/.flox`.
I think it would be best we kept using that directory for consistency with the other env types.
Also, `TempDir` _will_ be cleaned up eventually by the system, and generated again for every call to cache_path

```suggestion
        self.inner.path()
```

Reply:
My intention was generating it for every call to `flox activate -r`. I would expect `flox activate -r` in two different shells to be isolated from each other. Or if I run `flox activate -r -- touch $FLOX_ENV_CACHE/file`, I was thinking a second `flox activate -r` shouldn't be able to access the file. Otherwise there's state that I can't really see, but it  could mean I'll get a different result than someone else.

@ghudgins any thoughts?

Reply:
My understanding was that (aside from the performance aspect) we made remote environments use a common (virtual) .flox such that they share the same state.
I.e an install in one terminal would immediately take effect in the other.
IMO they should also share the same cache then.
Additionally, if you are running an environment that needs/builds caches in your rcfiles, you'd not want that to happen every time you open a new terminal.
Lastly, i understand that we try to make all envs behave uniformely, exceptions like this kinda break that assumption

Reply:
Discussed, I think we can resolve this now?

Reply:
Discussed, yes -- ephemeral envs per activation
It would still be nice to have these temp things in a flox managed location (i.e. in `flox.temp_dir` instead of a foreign location created by `TempDir`.
Can we add that to the `RemoveEnvironment` struct at creation and return those here?

Reply:
Hrm yeah that's the issue with exec now.. 
Ok, well these dots are ephemeral anyway so if anything changes we can come back without breaking things.